posts/intro-to-mathport.md

```diff
@@ -0,0 +1,224 @@
+without having to run it yourself.
+
+I'll described that setup up below, but first explain what sort of efforts are probably most useful right now to help the `mathlib` port.
+
+They are approximately in priority order, in terms of my guess about what will hold up the port the most.
+
+* Porting missing tactics from `mathlib3` to `mathlib4`.
+  This is still a huge task, and will not be automated in any way.
+  If you've contributed tactics to `mathlib3`, please consider trying to port them to Lean 4.
+  If you're interested in learning some Lean 4 metaprogramming, what better way to do that than porting tactics?
+  We'll write more about this in a future post, with some pointers about places to get started,
+  and how to hook up new tactic implementations to the existing tactic parsers that Mario has already ported to `mathlib4`.
+* Resolve outstanding [issues](https://github.com/leanprover-community/mathport/issues) in `mathport`.
+  (On some issues there's already an indicated fix but it needs implementing/testing.
+  Other issues still need diagnosis.)
+* Open the `mathlib3port` repository (instructions below), look at files (probably starting with "low-level" files),
+  and identify things that `mathport` should be doing better. Check there isn't already an open issue, then open an issue.
+  * Note: at this stage I think it would be a bad idea to actually take a file from `mathlib3port`, clean it up, and PR it to `mathlib4`.
+    That will hopefully come later, but we need to fix many `mathport` issues first.
+  * Instead, it is fine to make changes *that you think `mathport` should be doing already* and committing these on a branch,
+    so that you can link to diffs when opening `mathport` issues.
+  * For now, don't worry too much about the state of proofs.
+    We'd like to get to a state where the vast majority of *statements* are correctly translated as soon as possible.
+  * There are still many alignment problems between Lean 3 / `mathlib3` declarations, and Lean 4 / `mathlib4` declarations.
+    Sometimes these can be fixed by adding `#align` commands in `mathlib4`. Sometimes they may turn out to be `mathport` bugs.
+    Sometimes they will reflect deeper design problems we're going to need to talk about!
+
+
+# Background: what is `mathport`?
+
+`mathport` consists of two loosely coupled components: `binport` (largely Daniel's work), and `synport` (largely Mario's work).
+
+* `binport` constructs Lean 4 `.olean` files, from Lean 3 `.olean` files.
+  It largely works, and means that you can import `mathlib3` content into Lean 4 (as long as you don't expect to have source files!)
+  This is what lets us do things like:
+```
+import Mathbin
+
+#lookup3 algebraic_geometry.Scheme
+#check AlgebraicGeometry.Scheme
+```
+  Yay, Lean 4 has schemes! :-)
+  To see this file in action, you should check out a copy of the `mathlib3port` repository described below,
+  and make a new file there.
```

Comment:
To make the code be under the bullet point:
```suggestion
* `binport` constructs Lean 4 `.olean` files, from Lean 3 `.olean` files.
  It largely works, and means that you can import `mathlib3` content into Lean 4 (as long as you don't expect to have source files!)
  This is what lets us do things like:
  ```
  import Mathbin
  
  #lookup3 algebraic_geometry.Scheme
  #check AlgebraicGeometry.Scheme
  ```
  Yay, Lean 4 has schemes! :-)
  To see this file in action, you should check out a copy of the `mathlib3port` repository described below,
  and make a new file there.
```